コメント機能の追加、ユーザー画像の表示機能の追加

pokotyan opened the pull request:
# WHAT

コメント機能の追加
ユーザーの画像用のuploaderの作成、およびユーザー画像の表示機能の追加
# WHY

投稿されたprototypeに対するユーザーの反応を事前に確認することで、prototypeのブラッシュアップを図る。

pokotyan committed:
create pull request

pokotyan committed:
add comment_model

pokotyan committed:
add comments_controller

pokotyan committed:
add show_action to prototypes_controller

pokotyan committed:
add asociation to prototype_model,user_model

pokotyan committed:
change show_view to use ajax

pokotyan committed:
add routing for comments_create

pokotyan committed:
add action to CommentsController

pokotyan committed:
add js for ajax

pokotyan committed:
add partial_file for comment preview

pokotyan committed:
fix for text_field_tag

pokotyan committed:
fix partial_file for comment_count

前回までのコミットの状態では、コメント０件の状態でコメントを投稿しても、コメントのカウントが０のままで変わらなかった。
コメントが１件以上すでにある状態でshowページにアクセスした時のみコメントカウントのajaxが機能していた。
原因としては、if @comments.blank?　の判定がshowページにアクセスした時のみしか実行されてなかったため。
なのでパーシャルの中にコメントのカウントの判定を含めることでajaxが実行されるたびにコメント合計数の集計を行うように修正。
この修正により、コメント０件の状態でコメントを投稿したらちゃんとコメントのカウントが１になった。

KoichiOzaki commented on app/controllers/comments_controller.rb:
```diff
@@ -0,0 +1,16 @@
+    @comment = current_user.comments.build(create_params)
+    @comment.save
+    @prototype = @comment.prototype
+    @comments = Comment.where(prototype_id:@prototype)
```
@commentsという変数は必要でしょうか？
Prototypeモデルとアソシエーションしているので、@prototype.commentsで呼び出し可能なはずです。

pokotyan replied:
@prototype.commentsで呼び出すようにしました。

KoichiOzaki commented on app/controllers/prototypes_controller.rb:
```diff
@@ -8,6 +8,9 @@ def index
     @sub_images = @prototype.images.sub
+
+    @comment = Comment.new
+    @comments = Comment.where(prototype_id:@prototype)
```
@prototype.commentsで呼び出し可能なはずなので、変数は作らなくて良いです。

pokotyan replied:
@prototype.commentsで呼び出し、変数は作らないようにしました。

KoichiOzaki commented on app/views/comments/_comment.html.haml:
```diff
@@ -0,0 +1,15 @@
+    .media
+      .media-left
+        %a{href: "#"}
+          %img.media-object{alt: "64x64", "data-holder-rendered": "true", "data-src": "holder.js/64x64", src: "data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9InllcyI/PjxzdmcgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB3aWR0aD0iNjQiIGhlaWdodD0iNjQiIHZpZXdCb3g9IjAgMCA2NCA2NCIgcHJlc2VydmVBc3BlY3RSYXRpbz0ibm9uZSI+PGRlZnMvPjxyZWN0IHdpZHRoPSI2NCIgaGVpZ2h0PS ...
```
ここではユーザのプロフィール写真を読み込むようにしましょう

pokotyan replied:
ユーザーのプロフィール写真を表示するようにしました。

KoichiOzaki commented on app/views/comments/_comment.html.haml:
```diff
@@ -0,0 +1,15 @@
+  Comments
+  %span.num#comment_count (#{prototype.comments.count})
+.col-md-8
+  -comments.each do |comment|
```
部分テンプレート化するなら、繰り返し部分を部分テンプレート化してあげた方がいいです。
コメント欄のヘッダー部分

```
%h4.col-md-12
   Comments
   %span.num#comment_count (#{prototype.comments.count})
```

についてもここのまま部分テンプレート化しておいてもいいですが、_comments_header.html.hamlとかが適切だと思います。
（併せてrender元のビューファイルも変更が必要です）

pokotyan replied:
繰り返し部分のみrenderで回すように変更しました。

pokotyan committed:
fix partial_file

パーシャルの中身は繰り返し部分のみにするよう修正
comments#createアクションで変数を作りすぎてたのを必要最低限の変数のみ作るように修正

pokotyan committed:
make avatar_uploader

pokotyan committed:
add mount_uploader to user_model

pokotyan committed:
add avatar_column to users_table

pokotyan committed:
add parameter to registration/new for avatar

pokotyan committed:
add parameter to users#edit for avatar

pokotyan committed:
dispalay avatar to comment,prototype#show

pokotyan committed:
Unnecessary variable delete

@commentsは@prototype.commentsで呼び出すように変更

pokotyan committed:
display avatar to users#show and fix current_user

ユーザーページでavatarを表示するように設定。
また、current_userになってたので@userに変更（ログインユーザーが別ユーザーのshowページに訪問した時に間違った情報になるため）

KoichiOzaki commented:
[![LGTM](https://lgtm.in/p/oIm07N66q)](https://lgtm.in/i/oIm07N66q)
